rs/bitcoin/checker/tests/tests.rs

```diff
@@ -701,6 +754,23 @@ fn test_check_transaction_error() {
     let actual_cost = cycles_before - cycles_after;
     assert!(actual_cost > expected_cost);
     assert!(actual_cost - expected_cost < UNIVERSAL_CANISTER_CYCLE_MARGIN);
+
+    // Check metrics
+    MetricsAssert::from_querying_metrics(&setup).assert_contains_metric_matching(
+        r#"btc_check_requests_total\{type=\"check_transaction\"\} 5 \d+"#,
+    );
+    MetricsAssert::from_querying_metrics(&setup).assert_contains_metric_matching(
+        r#"btc_checker_http_calls_total\{provider=\"[a-z.]*\",status=\"500\"\} 1 \d+"#,
+    );
+    MetricsAssert::from_querying_metrics(&setup).assert_contains_metric_matching(
+        r#"btc_checker_http_calls_total\{provider=\"[a-z.]*\",status=\"200\"\} 1 \d+"#,
+    );
+    MetricsAssert::from_querying_metrics(&setup).assert_contains_metric_matching(
+        r#"btc_checker_http_calls_total\{provider=\"[a-z.]*\",status=\"404\"\} 1 \d+"#,
+    );
+    MetricsAssert::from_querying_metrics(&setup).assert_contains_metric_matching(
+        r#"btc_checker_http_calls_total\{provider=\"[a-z.]*\",status=\"SysTransient\"\} 1 \d+"#,
+    );
```

Comment:
nit: I would chain those, since every `MetricsAssert::from_querying_metrics(&setup)` is making an HTTP get request to get the metrics and the API is fluent by design.